compiler/rustc_resolve/src/effective_visibilities.rs

```diff
@@ -141,9 +142,10 @@ impl<'r, 'a, 'tcx> EffectiveVisibilitiesVisitor<'r, 'a, 'tcx> {
 
                     parent_id = ParentId::Import(binding);
                     binding = nested_binding;
+                    warn_ambiguity |= nested_binding.warn_ambiguity;
                 }
-
-                if binding.ambiguity.is_none()
+                let no_ambiguity = binding.ambiguity.is_none() || warn_ambiguity;
+                if no_ambiguity
```

Comment:
The `binding.ambiguity.is_some()` condition a few lines above also needs to be replaced with `binding.ambiguity.is_some() && !warn_ambiguity`.